build.c

```diff
@@ -610,6 +672,13 @@ build(void)
 			next = i;
 			if (jobs[i].failed)
 				++numfail;
+			/* must return the GNU/tokens once a job is done */
+			if (buildopts.gmakepipe[1] > 0 && gmakelatest != gmaketokens) {
+				if (write(buildopts.gmakepipe[1], --gmakelatest, 1) < 0) {
```

Comment:
I'm not sure if it matters, but this might not be the same job token that we acquired to start this job. Maybe we should track the job token in the job struct?

Reply:
I think we can shuffle the tokens around. In fact, despite documentation saying "don’t assume that all tokens are the same character", in my testing they usually are.